packages/frames.js/src/getFrameFlattened.ts

```diff
@@ -1,12 +1,29 @@
 import { version as framesjsVersion } from "../package.json";
 import type { Frame, FrameFlattened } from "./types";
 
+export function getFrameFlattened(
+  frame: Frame,
+  overrides?: Partial<FrameFlattened>
+): FrameFlattened;
+export function getFrameFlattened(
+  frame: Partial<Frame>,
+  overrides?: Partial<FrameFlattened>
+): Partial<FrameFlattened>;
+
 /**
  * Takes a `Frame` and formats it as an intermediate step before rendering as html
- * @param frame - The `Frame` to flatten
  * @returns a plain object with frame metadata keys and values according to the frame spec, using their lengthened syntax, e.g. "fc:frame:image"
  */
-export function getFrameFlattened(frame: Frame): FrameFlattened {
+export function getFrameFlattened(
+  /**
+   * The frame to flatten
+   */
+  frame: Partial<Frame>,
+  /**
+   * Optional overrides to apply to the frame
+   */
+  overrides?: Partial<FrameFlattened>
```

Comment:
i'm tempted to argue that this should be any object to allow the user to also specify custom properties. what do you think?

Reply:
If they use typescript it doesn't really make that much sense since the result type will complain about extra properties. I don't think that is necessary since they can achieve that without us handling us on type level.